operator/pkg/ingress/ingress.go

```diff
@@ -0,0 +1,447 @@
+		return err
+	}
+	return nil
+}
+
+func (ic *IngressController) handleIngressServiceUpdatedEvent(ingressServiceUpdated ingressServiceUpdatedEvent) error {
+	service := ingressServiceUpdated.ingressService
+	serviceKey, err := cache.MetaNamespaceKeyFunc(service)
+	if err != nil {
+		return err
+	}
+	ingressKey := getIngressKeyForService(service)
+	ingress, err := ic.getByKey(ingressKey)
+	if err != nil {
+		log.WithError(err).WithField("service", serviceKey).Warn("Failed to lookup ingress for service")
+		return err
+	}
+	scopedLog := log.WithField("ingress", ingressKey).WithField("service", serviceKey).
+		WithField("svc-status", service.Status.LoadBalancer).
+		WithField("ingress-status", ingress.Status.LoadBalancer)
+
+	if reflect.DeepEqual(service.Status.LoadBalancer, ingress.Status.LoadBalancer) {
+		scopedLog.Info("No change in load balancer status")
+		return nil
+	}
+
+	newIngressStatus := getIngressForStatusUpdate(ingress, service.Status.LoadBalancer)
+	_, err = k8s.Client().NetworkingV1().Ingresses(ingress.Namespace).UpdateStatus(context.Background(), newIngressStatus, metav1.UpdateOptions{})
+	if err != nil {
+		scopedLog.WithError(err).Warn("Failed to update ingress status")
+		return err
+	}
+	scopedLog.Info("Updated ingress status")
+	return nil
+}
+
+func toV1TypeMeta(in slim_metav1.TypeMeta) metav1.TypeMeta {
+	return metav1.TypeMeta{
+		Kind:       in.Kind,
+		APIVersion: in.APIVersion,
+	}
+}
+
+func toV1ObjectMeta(in slim_metav1.ObjectMeta) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:            in.Name,
+		Namespace:       in.Namespace,
+		ResourceVersion: in.ResourceVersion,
+		UID:             in.UID,
+		Labels:          in.Labels,
+		Annotations:     in.Annotations,
+	}
+}
+
+func getIngressForStatusUpdate(slimIngress *slim_networkingv1.Ingress, lb slim_corev1.LoadBalancerStatus) *networkingv1.Ingress {
+	slimIngressCopy := slimIngress.DeepCopy()
+	return &networkingv1.Ingress{
+		TypeMeta:   toV1TypeMeta(slimIngressCopy.TypeMeta),
+		ObjectMeta: toV1ObjectMeta(slimIngressCopy.ObjectMeta),
+		Status: networkingv1.IngressStatus{
+			LoadBalancer: corev1.LoadBalancerStatus{
+				Ingress: k8s.ConvertToK8sV1LoadBalancerIngress(lb.Ingress),
+			},
+		},
+	}
+}
+
+func (ic *IngressController) handleEvent(event interface{}) error {
+	var err error
+	switch ev := event.(type) {
+	case ingressAddedEvent:
+		log.WithField("ingress", ev.ingress.Name).Info("Handling ingress added event")
+		err = ic.handleIngressAddedEvent(ev)
+	case ingressUpdatedEvent:
+		log.WithField("ingress", ev.newIngress.Name).Info("handling ingress updated event")
+		err = ic.handleIngressUpdatedEvent(ev)
+	case ingressDeletedEvent:
+		log.WithField("ingress", ev.ingress.Name).Info("Handling ingress deleted event")
+		err = ic.handleIngressDeletedEvent(ev)
+	case ingressServiceUpdatedEvent:
+		log.WithField("service", ev.ingressService.Name).Info("Handling ingress service updated event")
```

Comment:
Let's change these log fields to `Debug` instead, otherwise the logs will be too verbose.